File: This_Self/This_Self.java
class Counter {
    private int value;

    Counter(int value) {
        this.value = value;
    }

    Counter increment() {
        this.value++;
        return this;
    }

    void print() {
        System.out.println("Value: " + this.value);
    }
}

public class This_Self {
    public static void main(String[] args) {
        new Counter(0).increment().increment().print();
    }
}

File: This_Self/This_Self.py
class Counter:
    def __init__(self, value: int) -> None:
        self.value = value

    def increment(self) -> "Counter":
        self.value += 1
        return self

    def print_value(self) -> None:
        print(f"Value: {self.value}")


def main() -> None:
    Counter(0).increment().increment().print_value()


if __name__ == "__main__":
    main()
